core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -314,7 +320,11 @@ contract PricelessPositionManager is FeePayer {
      * @notice After expiration of the contract the DVM is asked what for the prevailing price at the time of
      * expiration. In addition, pay the final fee at this time. Once this has been resolved token holders can withdraw.
      */
-    function expire() external onlyPostExpiration() fees() {
+    // TODO: what are the implications of calling this multiple times? can this not drain the contract by paying
+    // the final fee over and over again?
+    function expire() external onlyPostExpiration() onlyOpenState() fees() {
```

Comment:
Great catch here and defense with `onlyOpenState()` modifier. Is this comment out of date now?

Reply:
@chrismaree Can you add a test for future protection that ensures that `expire()` can only be called once on a post-expiry contract?

Reply:
Sure but I did not want to add that to this PR. I created an issue where we can test it explicitly. there are a number of cases where the same test needs to be done. See issue #989